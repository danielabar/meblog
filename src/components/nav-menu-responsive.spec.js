import React from "react"
import { render, fireEvent, screen } from "@testing-library/react"
import "@testing-library/jest-dom"

import NavMenuResponsive from "./nav-menu-responsive"

describe("NavMenuResponsive", () => {
  it("renders closed by default", () => {
    const container = render(<NavMenuResponsive />)
    expect(container).toMatchSnapshot()

    expect(screen.getByTestId("nav-menu-responsive-menu")).toHaveTextContent(
      "menu"
    )
    expect(
      screen.queryByTestId("nav-menu-responsive-close")
    ).not.toBeInTheDocument()
  })

  it("opens", () => {
    const container = render(<NavMenuResponsive />)
    fireEvent.click(screen.getByTestId("nav-menu-responsive-menu"))
    expect(container).toMatchSnapshot()

    expect(screen.getByTestId("nav-menu-responsive-close")).toHaveTextContent(
      "close"
    )
    expect(
      screen.queryByTestId("nav-menu-responsive-menu")
    ).not.toBeInTheDocument()
  })

  it("closes after being opened", () => {
    render(<NavMenuResponsive />)
    fireEvent.click(screen.getByTestId("nav-menu-responsive-menu"))
    fireEvent.click(screen.getByTestId("nav-menu-responsive-close"))

    expect(screen.getByTestId("nav-menu-responsive-menu")).toHaveTextContent(
      "menu"
    )
    expect(
      screen.queryByTestId("nav-menu-responsive-close")
    ).not.toBeInTheDocument()
  })
})
