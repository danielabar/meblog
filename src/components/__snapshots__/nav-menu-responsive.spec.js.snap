// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenuResponsive opens 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navOpen"
      >
        <button
          class="menuButton menuButtonClose"
          data-testid="nav-menu-responsive-close"
        >
          <svg
            fill="currentColor"
            height="1.7rem"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1.7rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <br />
          close
        </button>
        <ul
          class="navList"
        >
          <li />
          <li
            class="headerItem navItem"
          >
            <a
              class="navLink"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="headerItem navItem"
          >
            <a
              class="navLink"
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li
            class="headerItem navItem"
          >
            <a
              class="navLink"
              href="/about"
            >
              About
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navOpen"
    >
      <button
        class="menuButton menuButtonClose"
        data-testid="nav-menu-responsive-close"
      >
        <svg
          fill="currentColor"
          height="1.7rem"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1.7rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <br />
        close
      </button>
      <ul
        class="navList"
      >
        <li />
        <li
          class="headerItem navItem"
        >
          <a
            class="navLink"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="headerItem navItem"
        >
          <a
            class="navLink"
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li
          class="headerItem navItem"
        >
          <a
            class="navLink"
            href="/about"
          >
            About
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavMenuResponsive renders closed by default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav"
        data-testid="nav-menu-responsive"
      >
        <ul
          class="navList"
        >
          <li>
            <button
              class="menuButton"
              data-testid="nav-menu-responsive-menu"
            >
              <svg
                fill="currentColor"
                height="1.7rem"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1.7rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
              <br />
              menu
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav"
      data-testid="nav-menu-responsive"
    >
      <ul
        class="navList"
      >
        <li>
          <button
            class="menuButton"
            data-testid="nav-menu-responsive-menu"
          >
            <svg
              fill="currentColor"
              height="1.7rem"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1.7rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <br />
            menu
          </button>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
